cmd/swarm/run_test.go

```diff
@@ -179,16 +177,35 @@ func (c *testCluster) Stop() {
 
 func (c *testCluster) StartNewNodes(t *testing.T, size int) {
 	c.Nodes = make([]*testNode, 0, size)
+
+	errors := make(chan error, size)
+	nodes := make(chan *testNode, size)
 	for i := 0; i < size; i++ {
-		dir := filepath.Join(c.TmpDir, fmt.Sprintf("swarm%02d", i))
-		if err := os.Mkdir(dir, 0700); err != nil {
-			t.Fatal(err)
-		}
+		go func(nodeIndex int) {
+			dir := filepath.Join(c.TmpDir, fmt.Sprintf("swarm%02d", nodeIndex))
+			if err := os.Mkdir(dir, 0700); err != nil {
+				errors <- err
+				return
+			}
 
-		node := newTestNode(t, dir)
-		node.Name = fmt.Sprintf("swarm%02d", i)
+			node := newTestNode(t, dir)
```

Comment:
you're starting `<size>` nodes here, every one of them getting a separate process, also, since most of them would be started at the same time, we might be seeing more problems with the discovery tests (if that problem isn't fixed). why do we need parallelism here? and how will this react when you need to start `128`/`256` nodes at the same time?

Reply:
> How will this react when you need to start 128/256 nodes at the same time?

Node startup takes a long time. The parallelization reduces the test run time by 50% in case of 3 nodes. As I want to remove the global (start only once) cluster this improvement balances that performance loss.

I'm not sure the number of nodes we can handle when they are starting in parallel. Let's fix that problem when we get there. As we currently start 1 or 3 nodes with this function. At that time - when we need hundreds of node starting -  we can limit the number of nodes starting in parallel.

Reply:
test this on Travis and report your findings. Again, your local machine specs are incomparable